internal/pkg/manifest/env.go

```diff
@@ -41,6 +69,45 @@ type environmentVPCConfig struct {
 	Subnets subnetsConfiguration `yaml:"subnets,omitempty"`
 }
 
+func (cfg *environmentVPCConfig) loadVPCConfig(vpc *config.CustomizeEnv) {
+	if vpc.IsEmpty() {
+		return
+	}
+	if adjusted := vpc.VPCConfig; adjusted != nil {
+		cfg.loadAdjustedVPCConfig(adjusted)
+	}
+	if imported := vpc.ImportVPC; imported != nil {
+		cfg.loadImportedVPCConfig(imported)
+	}
+}
+
+func (cfg *environmentVPCConfig) loadAdjustedVPCConfig(vpc *config.AdjustVPC) {
+	cfg.CIDR = ipNetP(vpc.CIDR)
+	cfg.Subnets.Public = make([]subnetConfiguration, len(vpc.PublicSubnetCIDRs))
+	cfg.Subnets.Private = make([]subnetConfiguration, len(vpc.PrivateSubnetCIDRs))
+	hasCustomAZNames := len(vpc.AZs) > 0
+	for i := range vpc.PublicSubnetCIDRs {
```

Comment:
will there always be the same number of public and private subnets?

Reply:
+1

Reply:
That's the case today: 
https://github.com/aws/copilot-cli/blob/c94e2b4e83e663a2a322e18e6bf9bd119b9566d6/internal/pkg/cli/validate.go#L815

The user first chooses the number of AZs, and then they have to specify the CIDR ranges for each subnet type in the AZs. 

But I agree that it'd be safer to write the method without this assumption. I updated the flow **with a more flexible assumption** that the number of AZs selected >= number of CIDR ranges specified. I think that's safer.